### PR TITLE
Install efs-utils from github

### PR DESCRIPTION
**Pull request by kbasv**
**Is this a bug fix or adding new feature?**
New feature

**What is this PR about? / Why do we need it?**
Bump efs-utils to `1.30.2-1`. Install latest efs-utils from github since it is not available on yum . Additionally, provide an option to install efs-utils from yum repo.

**What testing is done?** 
Manual tests on my cluster.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -32,7 +32,10 @@ ENV EFS_CLIENT_SOURCE=$client_source
 RUN GOOS=${TARGETOS} GOARCH=${TARGETARCH} make aws-efs-csi-driver
 
 FROM amazonlinux:2.0.20210219.0
-RUN yum install amazon-efs-utils-1.30.1-1.amzn2.noarch -y
+# Install efs-utils from github
+RUN yum -y install git rpm-build make
+RUN git clone https://github.com/aws/efs-utils
+RUN cd efs-utils && make rpm && yum -y install build/amazon-efs-utils*rpm
```
Comment (jqmichael) [inline Dockerfile:38]: Do we want to make this behavior configurable, e.g. from an `ARG`, like what we do for `client_source`?

**Comment by kbasv** [comment]
Flaky e2e Tests.

```
go test ./test/e2e/ -v -kubeconfig=$HOME/.kube/config --region=us-east-1 --report-dir="./results" -ginkgo.focus="should continue reading/writing without hanging" --cluster-name="eks-test" --file-system-id fs-92107410
W0505 10:40:57.696449   20837 test_context.go:414] Unable to find in-cluster config, using default host : http://127.0.0.1:8080
I0505 10:40:57.696605   20837 test_context.go:423] Tolerating taints "" when considering if nodes are ready
May  5 10:40:57.696: INFO: The --provider flag is not set. Continuing as if --provider=skeleton had been used.
=== RUN   TestEFSCSI
2021-05-05 10:40:57.696778 I | Starting e2e run "23002d2a-ff97-4265-8119-33bf009875a9" on Ginkgo node 1
Running Suite: EFS CSI Suite
============================
Random Seed: 1620225657 - Will randomize all specs
Will run 1 of 158 specs

STEP: Using already-created EFS file system "fs-92107410"
SSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSS
------------------------------
[efs-csi] EFS CSI [Driver: efs.csi.aws.com] 
  should continue reading/writing without hanging after the driver pod is restarted
  /Users/kb/go/aws-efs-csi-driver/test/e2e/e2e.go:255
[BeforeEach] [efs-csi] EFS CSI
  /Users/kb/go/pkg/mod/k8s.io/kubernetes@v1.18.10/test/e2e/framework/framework.go:178
STEP: Creating a kubernetes client
May  5 10:40:57.697: INFO: >>> kubeConfig: /Users/kb/.kube/config
STEP: Building a namespace api object, basename efs
May  5 10:40:59.083: INFO: Found PodSecurityPolicies; assuming PodSecurityPolicy is enabled.
May  5 10:40:59.134: INFO: Found ClusterRoles; assuming RBAC is enabled.
STEP: Binding the e2e-test-privileged-psp PodSecurityPolicy to the default service account in efs-8081
STEP: Waiting for a default service account to be provisioned in namespace
[It] should continue reading/writing without hanging after the driver pod is restarted
  /Users/kb/go/aws-efs-csi-driver/test/e2e/e2e.go:255
STEP: Creating efs pvc & pv
STEP: Creating pod on node "ip-192-168-57-78.ec2.internal" to mount pvc "efs-8081" and run "touch /mnt/volume1/efs-8081-2021-05-05T10:40:59-04:00 && trap exit TERM; while true; do sleep 1; done"
STEP: Getting driver pod on node "ip-192-168-57-78.ec2.internal"
STEP: Deleting driver pod "efs-csi-node-spzs7" on node "ip-192-168-57-78.ec2.internal"
May  5 10:41:39.405: INFO: Deleting pod "efs-csi-node-spzs7" in namespace "kube-system"
May  5 10:41:39.423: INFO: Wait up to 5m0s for pod "efs-csi-node-spzs7" to be fully deleted
STEP: Execing a write via the pod on node "ip-192-168-57-78.ec2.internal"
May  5 10:41:41.445: INFO: ExecWithOptions {Command:[/bin/sh -c touch /mnt/volume1/efs-8081-2021-05-05T10:41:41-04:00] Namespace:efs-8081 PodName:pvc-tester-5glqk ContainerName:write-pod Stdin:<nil> CaptureStdout:true CaptureStderr:true PreserveWhitespace:false}
May  5 10:41:41.445: INFO: >>> kubeConfig: /Users/kb/.kube/config
May  5 10:41:44.962: INFO: verified exec in pod succeeded
[AfterEach] [efs-csi] EFS CSI
  /Users/kb/go/pkg/mod/k8s.io/kubernetes@v1.18.10/test/e2e/framework/framework.go:179
May  5 10:41:44.985: INFO: Waiting up to 3m0s for all (but 0) nodes to be ready
STEP: Destroying namespace "efs-8081" for this suite.

• [SLOW TEST:47.313 seconds]
[efs-csi] EFS CSI
/Users/kb/go/aws-efs-csi-driver/test/e2e/e2e.go:211
  [Driver: efs.csi.aws.com]
  /Users/kb/go/aws-efs-csi-driver/test/e2e/e2e.go:219
    should continue reading/writing without hanging after the driver pod is restarted
    /Users/kb/go/aws-efs-csi-driver/test/e2e/e2e.go:255
------------------------------
SSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSS
JUnit report was created: /Users/kb/go/aws-efs-csi-driver/test/e2e/results/junit_01.xml

Ran 1 of 158 Specs in 47.314 seconds
SUCCESS! -- 1 Passed | 0 Failed | 0 Pending | 157 Skipped
--- PASS: TestEFSCSI (47.32s)
PASS
ok  	github.com/kubernetes-sigs/aws-efs-csi-driver/test/e2e	48.443s

```

**Comment by kbasv** [comment]
Retry e2e after marking restart tests as disruptive

File: Dockerfile (hunk 1 above)
Comment (kbasv) [reply Dockerfile]: Made it configurable

File: Dockerfile [hunk 2]
```diff
@@ -25,14 +25,24 @@ RUN echo "I am running on $(uname -s)/$(uname -m)"
...
 ARG client_source=k8s
 ENV EFS_CLIENT_SOURCE=$client_source
 
 RUN GOOS=${TARGETOS} GOARCH=${TARGETARCH} make aws-efs-csi-driver
 
 FROM amazonlinux:2.0.20210219.0
-RUN yum install amazon-efs-utils-1.30.1-1.amzn2.noarch -y
+# Install efs-utils from github by default. It can be overriden to `yum` with --build-arg when building the Docker image.
+# If value of `EFSUTILSSOURCE` build arg is overriden with `yum`, docker will install efs-utils from Amazon Linux 2's yum repo.
+ARG EFSUTILSSOURCE=github
```
Comment (wongma7) [inline Dockerfile:37]: nit, to be consistent with EFS_CLIENT_SOURCE it should be underscore separated :p

File: Dockerfile [hunk 3]
```diff
@@ -25,14 +25,24 @@ RUN echo "I am running on $(uname -s)/$(uname -m)"
...
+RUN if [ "$EFSUTILSSOURCE" = "yum" ]; \
+    then echo "Installing efs-utils from Amazon Linux 2 yum repo" && \
+         yum -y install amazon-efs-utils-1.30.1-1.amzn2.noarch; \
+    else echo "Installing efs-utils from github" && \
+         yum -y install git rpm-build make && \
```
Comment (wongma7) [inline Dockerfile:42]: i think we should remove these packages afterward. I don't think it matters much in terms of size, but also we are constantly getting reports of CVE from packages in the image that we don't even depend on at runtime.

File: Dockerfile (hunk 2 above)
Comment (kbasv) [reply Dockerfile:37]: I was trying to be consistent with `TARGETOS` and `TARGETARCH`. I can make the arg `EFS_CLIENT_SOURCE`

**Pull request merged**